wyvern/components/models/modelbit_component.py

```diff
@@ -153,3 +123,68 @@ async def inference(self, input: MODEL_INPUT, **kwargs) -> MODEL_OUTPUT:
             data=output_data,
             model_name=self.name,
         )
+
+    async def build_requests(
+        self,
+        input: INPUT_TYPE,
+    ) -> Tuple[List[Identifier], List[Any]]:
+        """
+        This method builds requests for Modelbit. This method should be implemented by the subclass.
+        """
+        raise NotImplementedError
+
+
+class ModelbitComponent(
+    ModelbitMixin[MODEL_INPUT, MODEL_OUTPUT],
+    MultiEntityModelComponent[MODEL_INPUT, MODEL_OUTPUT],
+):
+    """
+    ModelbitComponent is a base class for all modelbit model components. It provides a common interface to implement
+    all modelbit models.
+
+    ModelbitComponent is a subclass of ModelComponent.
+
+    Attributes:
+        AUTH_TOKEN: A class variable that stores the auth token for Modelbit.
+        URL: A class variable that stores the url for Modelbit.
+    """
+
+    async def build_requests(
+        self,
+        input: MODEL_INPUT,
+    ) -> Tuple[List[Identifier], List[Any]]:
+        """
+        Please refer to modlebit batch inference API:
+            https://doc.modelbit.com/deployments/rest-api/
+        """
+        target_entities: List[
+            Union[WyvernEntity, BaseWyvernRequest]
+        ] = input.entities or [input.request]
+        target_identifiers = [entity.identifier for entity in target_entities]
+        all_requests = [
+            [
+                idx + 1,
+                [
+                    self.get_feature(identifier, feature_name)
+                    for feature_name in self.modelbit_features
+                ],
+            ]
+            for idx, identifier in enumerate(target_identifiers)
+        ]
+        return target_identifiers, all_requests
+
+
+class SingleEntityModelbitComponent(
+    ModelbitMixin[REQUEST_ENTITY, MODEL_OUTPUT],
+    SingleEntityModelComponent[REQUEST_ENTITY, MODEL_OUTPUT],
+):
+    async def build_requests(
```

Comment:
I think we should make this `@final` so that it doesn't get overridden

Reply:
TIL. didn't know there's a final stuff in python